src/libcore/alloc/mod.rs

```diff
@@ -78,7 +81,10 @@ impl AllocInit {
         match self {
             AllocInit::Uninitialized => (),
             AllocInit::Zeroed => {
-                memory.ptr.as_ptr().add(offset).write_bytes(0, memory.size - offset)
+                // SAFETY: `offset` must be smaller than or equal to `memory.size`,
```

Comment:
Maybe mention "caller" here; seems like a useful keyword to drop each time requirements are just forwarded.